### PR TITLE
Modal: pass the closedDeferred promise to removeAfterAnimate function

### PR DESCRIPTION
**Pull request by armagedon**
If I close the modal the closed promise will not be called.
```javascript
modalInstance.closed.then(function() {
    console.log('closed', arguments);
});
```

### CODE REVIEW

**Comment by Foxandxss** [comment]
Sure, let me grab my crystal ball and figure out what is the purpose of this PR.

**Comment by armagedon** [comment]
Hallo,
I am little bit confused about the failed tests and about the implementation of the function toBeResolvedWith in modal.spec.js.

The promise closedDeferred will be resolved only in the function afterAnimating() in modal.js line 369.
I added console.log("afterAnimating", closedDeferred); in the function afterAnimating() (line 359).
Without my changes, I get follow output with log level DEBUG
> LOG: 'afterAnimating', undefined and the line closedDeferred.resolve(); will not be called.

With my changes, I get 
> LOG: 'afterAnimating', Deferred{promise: Promise{$$state: Object{status: ...}}, resolve: function (value) { ... }, reject: function (value) { ... }, notify: function (value) { ... }}

**Comment by Foxandxss** [comment]
I see, my friend @wesleycho is looking into it.

Thanks.

File: src/modal/modal.js [hunk 1]
```diff
@@ -302,7 +302,7 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap'])
           openedClasses.remove(modalBodyClass, modalInstance);
           appendToElement.toggleClass(modalBodyClass, openedClasses.hasKey(modalBodyClass));
           toggleTopWindowClass(true);
-        });
+        }, modalWindow.closedDeferred);
```
Comment (wesleycho) [inline src/modal/modal.js]: I think this is supposed to be `modalInstance.closedDeferred`

**Comment by wesleycho** [comment]
@armagedon looks like somehow the tests were flawed for this - I fixed them in #5326.

**Pull request closed without merging**